### ISSUE TITLE
BN update problem

### ISSUE THREAD

**Issue by chenxinhua** [yaojieliu/ECCV20-STDN#11]
in model/model.py, it should add  update_ops to update mean and variance of BN.

`def get_train_op(sum_loss, global_step, config, scope_name):
  update_ops = tf.get_collection(tf.GraphKeys.UPDATE_OPS)
  # Variables that affect learning rate.
  decay_steps = config.NUM_EPOCHS_PER_DECAY * config.STEPS_PER_EPOCH

  # Decay the learning rate exponentially based on the number of steps.
  lr = tf.train.exponential_decay(learning_rate=config.LEARNING_RATE,
                                  global_step=global_step,
                                  decay_steps=decay_steps,
                                  decay_rate=config.LEARNING_RATE_DECAY_FACTOR,
                                  staircase=True)
  # Generate moving averages of all losses and associated summaries.
  loss_averages = tf.train.ExponentialMovingAverage(0.9, name='avg')
  loss_averages_op = loss_averages.apply([sum_loss])
  # Compute gradients.
  with tf.control_dependencies([loss_averages_op]):
    opt = tf.train.AdamOptimizer(lr)
    grads = opt.compute_gradients(sum_loss, var_list=tf.trainable_variables(scope=scope_name))
  # Apply gradients.
  apply_gradient_op = opt.apply_gradients(grads, global_step=global_step)

  # Track the moving averages of all trainable variables.
  with tf.name_scope('TRAIN-'+scope_name) as scope:
    variable_averages = tf.train.ExponentialMovingAverage(config.MOVING_AVERAGE_DECAY, global_step)
    variables_averages_op = variable_averages.apply(tf.trainable_variables(scope=scope_name))

  with tf.control_dependencies([apply_gradient_op, variables_averages_op] + update_ops):
    train_op = tf.no_op(name='train-'+scope_name)

  return train_op, lr`

**Comment by Elijah-Yi** [comment]
in layers.batch_norm parameter: **updates_collections** can set **updates_collections=None** to force the updates in place
